src/LazyIndex.ts

```diff
@@ -0,0 +1,14 @@
+// This file is used as an entry point to register all components that can be lazy loaded.
```

Comment:
I do not think it's a good idea to show how to create a Lazy custom component.

The thing about lazy loading is it's useful only when you don't know ahead of time which component will ultimately end up in the search page. This is why we needed that in the Search UI.

Now, for someone doing a custom implementation, with custom component, I sure hope they at least know which custom component they will use ahead of time (because... well if they are not gonna use it in their page, why code it in the first place ?).

Otherwise, using lazy loading "for everything" is actually detrimental, and will slow things down, and add needless complexity.

I would probably remove everything that has to do with coding custom component to load in a lazy loading pattern.

I'm open to counter arguments :D

Reply:
> I'm open to counter arguments :D

Hehe.

If you really think this would cause more harm than good, I'm fine with not debating about it and removing the lazy part, it's not like a lot of work went into it ;)

But my main reason is "why not" :)

Why not let the dev decide if they need it or not? Sure, we could add comments to explain the drawbacks, but I like the idea of having a common ground for some stuff you _might_ want to do if you create custom components.

The feature is flashy, devs might want to support it for the sake of it, better start them on the right foot than having to deal with a flaky implementation later :)

Reply:
```
The feature is flashy, devs might want to support it for the sake of it, [...]
```

My point exactly... I'm afraid people will use it without understanding anyway, and then they get the impression that "using coveo is soooo complicated, it sucks etc..".

So that would be my "why not" answer.

Because you don't need that to implement Coveo, it's something that is relatively hard to understand (asynchronous loading of chunks, asynchronous initialization), and it has pretty much 0 advantage.

I think we have to remember some people interacting with our stuff have difficulties appending a `div` into their page, or understand how a query selector works :)